lib/SIL/IR/TypeLowering.cpp

```diff
@@ -2447,38 +2447,27 @@ static CanAnyFunctionType getStoredPropertyInitializerInterfaceType(
 /// (property-type) -> backing-type.
 static CanAnyFunctionType getPropertyWrapperBackingInitializerInterfaceType(
                                                      TypeConverter &TC,
-                                                     VarDecl *VD) {
+                                                     VarDecl *VD,
+                                                     bool fromWrappedValue) {
   CanType resultType =
       VD->getPropertyWrapperBackingPropertyType()->getCanonicalType();
 
   auto *DC = VD->getInnermostDeclContext();
-  CanType inputType =
-    VD->getPropertyWrapperInitValueInterfaceType()->getCanonicalType();
-
-  auto sig = DC->getGenericSignatureOfContext();
-
-  AnyFunctionType::Param param(
-      inputType, Identifier(),
-      ParameterTypeFlags().withValueOwnership(ValueOwnership::Owned));
-  // FIXME: Verify ExtInfo state is correct, not working by accident.
-  CanAnyFunctionType::ExtInfo info;
-  return CanAnyFunctionType::get(getCanonicalSignatureOrNull(sig), {param},
-                                 resultType, info);
-}
-
-static CanAnyFunctionType getPropertyWrapperInitFromProjectedValueInterfaceType(TypeConverter &TC,
-                                                                                VarDecl *VD) {
-  CanType resultType =
-      VD->getPropertyWrapperBackingPropertyType()->getCanonicalType();
-
-  Type interfaceType = VD->getPropertyWrapperProjectionVar()->getInterfaceType();
-  if (interfaceType->hasArchetype())
-    interfaceType = interfaceType->mapTypeOutOfContext();
+  CanType inputType;
+  if (fromWrappedValue) {
+    inputType = VD->getPropertyWrapperInitValueInterfaceType()->getCanonicalType();
+  } else {
+    Type interfaceType = VD->getPropertyWrapperProjectionVar()->getInterfaceType();
+    if (interfaceType->hasArchetype())
```

Comment:
If VarDecl::getInterfaceType() returns something with an archetype it indicates a bug elsewhere, since most callers don't expect this to occur. Were you seeing this fail with closure parameters in generic contexts or something weird like that?

Reply:
This code was there before (I just combined two functions into one because everything after computing `inputType` is repeated). I don't remember why I added it and it might just be unnecessary - I'm running the tests locally now without it to see if anything breaks.